File: hsg/ĐỀ_TIN_HOC_NGUỒN/PHAN1/HSG_THPTAnLao_PHAN1_03/BAI1.pas
Program BAI1;
Var
	f1, f2 : Text;
	c1, c2 : char;
	count : LongInt;
Begin
	Assign(f1, 'BAI1.INP');
	Assign(f2, 'BAI1.OUT');
	Reset(f1);
	Rewrite(f2);
	
	Read(f1, c1);
	c2 := '@';
	count := 1;

	While not eof(f1) Do
	Begin
		Read(f1, c2);
		If (c2 = c1) Then
			count := count + 1
		Else
		Begin
			Write(f2, c1);
			If count > 1 Then Write(f2, count);
			c1 := c2;
			count := 1;
		End;
	End;

        Write(f2, c1);
        If count > 1 Then Write(f2, count);

	Close(f1);
	Close(f2);
End.

File: hsg/de_tin_nguon/PHAN1/HSG_THPTDoSon_phan1_01/Bai1.py
with open("Bai1.inp") as fi:
    n=int(fi.readline())
    s= fi.readline()
with open("Bai1.out",'w') as fo:
    s=list(map(''.join, zip(*[iter(s)]*n)))
    s=" ".join(s)
    fo.write(s + "\n")
    fo.write(str(s.count(" ")+1))
